modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -412,6 +412,26 @@ def test_loc(data):
     with pytest.raises(KeyError):
         modin_df.loc["NO_EXIST"]
 
+    # From issue #3764
+    pandas_df = pandas.DataFrame([[1, 2, 3], [4, 5, 6]])
+    modin_df = pd.DataFrame([[1, 2, 3], [4, 5, 6]])
+
+    pandas_df.loc[2] = pandas_df.loc[1]
+    modin_df.loc[2] = modin_df.loc[1]
+    df_equals(modin_df, pandas_df)
+
+    pandas_df.loc[6] = pandas_df.loc[1]
+    modin_df.loc[6] = modin_df.loc[1]
+    df_equals(modin_df, pandas_df)
+
+    pandas_df.loc[lambda df: 70] = pandas_df.loc[1]
+    modin_df.loc[lambda df: 70] = modin_df.loc[1]
+    df_equals(modin_df, pandas_df)
+
+    pandas_df.loc[90] = pandas_df.loc[70]
+    modin_df.loc[90] = modin_df.loc[70]
+    df_equals(modin_df, pandas_df)
```

Comment:
Please use `eval_general` as current form neither checks for exception equality nor correctly waits for async Modin calls to complete.
Something like
```suggestion
    for left, right in [(2, 1), (6, 1), (lambda df: 70, 1), (90, 70)]:

        def _test_loc(df):
            df.loc[left] = df.loc[right]
            return df

        eval_general(modin_df, pandas_df, _test_loc)
```

Reply:
You should get the correct behavior if you define `_test_loc` outside of the for loop in @vnlitvinov 's suggestion :)